tests/auto/qtpromise/warnings/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+qtpromise_add_test(warnings
+    SOURCES
+        tst_warnings.cpp
+)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    target_compile_options(qtpromise.tests.auto.warnings
```

Comment:
I tried to use `add_compile_options` but it didn't seem to work. For this test to work, we need to compile it with optimizations, however `qtpromise_add_test` function sets optimization level to -O0. What I don't like here is how I have to refer to a test by it's internal name.